app/iot/service.py

```diff
@@ -26,24 +27,32 @@ class IOTService:
     def __init__(self) -> None:
         self.devices: dict[str, Device] = {}
 
-    def register_device(self, device: Device) -> str:
-        device.connect()
+    async def register_device(self, device: Device) -> str:
+        await device.connect()
```

Comment:
Ensure that the `device.connect()` method is correctly implemented as an asynchronous function in the device classes. This is crucial for the `await` statement to work properly.